### PR TITLE
firefox-esr-91-unwrapped: drop

### PR DESCRIPTION
**Pull request by mweinelt**
The 91 series has reached its end of life.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
@ofborg eval

File: pkgs/applications/networking/browsers/firefox/common.nix [hunk 1]
```diff
@@ -228,7 +228,6 @@ buildStdenv.mkDerivation ({
     })
   ]
   ++ lib.optional (lib.versionAtLeast version "86") ./env_var_for_system_dir-ff86.patch
-  ++ lib.optional (lib.versionAtLeast version "90" && lib.versionOlder version "95") ./no-buildconfig-ffx90.patch
   ++ lib.optional (lib.versionAtLeast version "96") ./no-buildconfig-ffx96.patch
```
Comment (ajs124) [inline pkgs/applications/networking/browsers/firefox/common.nix:231]: some of these can be made unconditional now, but we can do that in a separate PR

**Review by ajs124 — APPROVED** [review]

**Comment by mweinelt** [comment]
Waiting for #191339 to bring nss 3.79.1 into master.

**Comment by vcunat** [comment]
What about merging to `staging-next` already?  Due to the nss update it won't build in there anyway.

**Comment by mweinelt** [comment]
Rebased.

**Comment by vcunat** [comment]
So, we won't add dummy attributes with `throw`?

**Comment by mweinelt** [comment]
No strong opinion on aliases. I added them for the stable release, we can also add them here to present users with a hint how to fix it.

**Comment by vcunat** [comment]
Same for me.  Sounds OK.
